### ISSUE TITLE
"Accrual Schedule" in BC

### ISSUE THREAD

**Issue by TabrejAlam3K** [3ktech/General#124]
Hi Audhi/Tabrej,

We require a seperate report/schedule for "Accruals" in BC (like that of prepayement schedule, but in a different format). Attaching the sample format of the report for your ref.

Pavan from my team, will be assisting you for any further clarifications.

[Accrual Schedule.xlsx](https://github.com/3ktech/General/files/10791008/Accrual.Schedule.xlsx)

**Comment by TabrejAlam3K** [comment]
@audhi3k - Today I will connect with Mr. Pavan and get the proper requirement.

**Comment by TabrejAlam3K** [comment]
@audhi3k - I have discussed with Mr. Pavan and get the requirement.

**Comment by audhi3k** [comment]
@TabrejAlam3K we will discuss this

**Comment by TabrejAlam3K** [comment]
@audhi3k - Need to sit with you when you feel free.

**Comment by TabrejAlam3K** [comment]
@audhi3k - Today we can sit and analyzing the requirement.

**Comment by TabrejAlam3K** [comment]
@audhi3k - I have discussed and show case of Account scheduled whatever we created for example but it is not use full for the client.

We need to customize the new reports based on their excel sheet and get the data from **General Ledger Entries** table.

**Comment by TabrejAlam3K** [comment]
Credit Amount = Accrual
Debit Amount = Accrual Reversal
Net Change = Credit - Debit

**Comment by TabrejAlam3K** [comment]
Need to discuss with Ms. @audhi3k for which technical need to work upon.

**Comment by TabrejAlam3K** [comment]
Need to work upon Field Mapping related to the report and forward to Ms. @audhi3k for Confirmation once she confirm then we can move this into the Technical Person.

**Comment by TabrejAlam3K** [comment]
Field Mapping is completed and we can sit with the technical to explain everything.

**Comment by TabrejAlam3K** [comment]
@audhi3k - Can you please align the technical person, so that we can connect and told about the requirement.

**Comment by sandhya3k** [comment]
@TabrejAlam3K  
Iam available to take the requirement. After the call we can connect.

**Comment by TabrejAlam3K** [comment]
@sandhya3k - Ok We will connect on after the internal session.

**Comment by sandhya3k** [comment]
@TabrejAlam3K 

 Iam not able to test the excel report. because There is no data in sandbox_india env, please create the data and let me know.

cc: @audhi3k

**Comment by TabrejAlam3K** [comment]
@sandhya3k Ok let me do some transaction on Sandbox environment.

CC : @audhi3k

**Comment by TabrejAlam3K** [comment]
@audhi3k - Mr. Pavan is on leave today and tomorrow.

**Comment by TabrejAlam3K** [comment]
As per our discussion Mr. Pavan will confirm after posting some entries related to Accruals.

**Comment by TabrejAlam3K** [comment]
@sandhya3k - Please share the report caption i am not able to find the Reports in Merit_Sandbox

**Comment by TabrejAlam3K** [comment]
@sandhya3k - Facing some error when i try to run the reports in below date format

Start Date - OD
End Date - Current Date

Error Snapshot. 

![Image](https://user-images.githubusercontent.com/116171982/226277513-9123c54a-5673-4e89-813a-8bc3802d760a.png)

CC : @audhi3k

**Comment by TabrejAlam3K** [comment]
@sandhya3k - Please update on client is asking about it. We need to dispatch to the client for testing.

CC : @audhi3k

**Comment by TabrejAlam3K** [comment]
@Mahesh3ktech - As per our discussion with Ms. Sandhya, Please update me on once you done you customization.

CC : @audhi3k @sandhya3k

**Comment by TabrejAlam3K** [comment]
@Mahesh3ktech - Please update the status of this.

**Comment by TabrejAlam3K** [comment]
@Mahesh3ktech - As per our discussion please update me once it will completed. We need to close this today.

CC : @audhi3k

**Comment by TabrejAlam3K** [comment]
@Mahesh3ktech - Please update on we need to close this immediately. Client is asking about this issues many times.

CC : @audhi3k

**Comment by Mahesh3ktech** [comment]
@TabrejAlam3K - Total value is not coming in correct place and opening balance and closing balance calculation is pending. ill complete these changes today.

**Comment by TabrejAlam3K** [comment]
@Mahesh3ktech - Please update on.

CC : @audhi3k

**Comment by Mahesh3ktech** [comment]
@TabrejAlam3K  - AS per requirement first scenario working fine and started second scenario and third scenario, Today ill complete report.

![Image](https://user-images.githubusercontent.com/110819826/228408969-167a7ec8-176b-4208-b0d2-64dd091875ce.png)


cc: @audhi3k

**Comment by TabrejAlam3K** [comment]
@Mahesh3ktech - If the value in not available in month column so, that should be hide in the output file. Currently it is showing as blank(January 2023, February 2023 and March 2023).

CC : @audhi3k

**Comment by Mahesh3ktech** [comment]
@TabrejAlam3K - Report development is completed, kindly test and update the status.

CC: @audhi3k

**Comment by TabrejAlam3K** [comment]
@Mahesh3ktech - I have tested and forwarded to the client for testing in Sandbox environment.

CC : @audhi3k

**Comment by TabrejAlam3K** [comment]
Hi Mr. @Mahesh3ktech - Have you completed the Accrual Reports.

1. Need to Change Report Caption - Accrual Schedule
2. G/L Account Name - It should be come from the related G/L Master and field name is **Name**
3. In Accrual Closing Balance - Opening Balance, Current Month Balance and

**Comment by Mahesh3ktech** [comment]
Hi @TabrejAlam3K - First two points completed and working on last point.

**Comment by TabrejAlam3K** [comment]
I have tested and forwarded to the Client, Waiting their comments on

CC : @audhi3k @niranjand22

**Comment by TabrejAlam3K** [comment]
@Mahesh3ktech - Report is not working properly.

Please test the report with date filter form 01-12-2022 to till date

We have two month data but it is still considering only one month data.

CC : @audhi3k

**Comment by TabrejAlam3K** [comment]
Hi Mr. @Mahesh3ktech - Have you completed it please completed it and let me ping.

CC : @audhi3k

**Comment by TabrejAlam3K** [comment]
Hi Mr. @Mahesh3ktech - Please update the current status of the report, Have you completed whatever the change required on this.
this task is already taken too much time please complete it on priority, client is asking many times about this reports.


CC : @audhi3k @niranjand22 @narayana3kt

**Comment by Mahesh3ktech** [comment]
Hi @TabrejAlam3K - Today ill check and close it.

**Comment by TabrejAlam3K** [comment]
@Mahesh3ktech - Mr. Mahesh how long it will get completed, Client ping me for this again and again.

CC : @audhi3k @Shasthara @narayana3kt

**Comment by Mahesh3ktech** [comment]
@TabrejAlam3K - I'm working on the changes.

**Comment by Mahesh3ktech** [comment]
@TabrejAlam3K - Changes has been done and Kindly check and update.

CC : @audhi3k

**Comment by TabrejAlam3K** [comment]
Hi Mr. @Mahesh3ktech - I am not able to find the reports in Merit Sandbox_Environment

![Image](https://user-images.githubusercontent.com/116171982/234479328-8abe9237-d4f6-417b-89ee-cb6950b6652c.png)

CC : @audhi3k @Shasthara @narayana3kt

**Comment by TabrejAlam3K** [comment]
I have already informed to the client User Mr. Pavan. He will test and confirm.

**Comment by niranjand22** [comment]
@TabrejAlam3K  Pleas4 update the recent status

**Comment by TabrejAlam3K** [comment]
See the below changes required on.
1. On Request Page required two more filter field
    1. GL Account No.
    2. Bal GL Account No.
2. In Accrual Table - GL Account No. should be Expanse Account and Bal GL Account should be fixed Accrual Account.
3. In Accrual Reversal Table - GL Account No. should be fixed Accrual Account and Bal GL Account should be Expanse Account.

@niranjand22 - If you required any clarification on please let me know.

**Comment by niranjand22** [comment]
@Mahesh3ktech  Hi pls let me know once 1 and 2nd point is done.. point 3 is already working fine.
Connect with me for any clarification required

**Comment by Mahesh3ktech** [comment]
Hi @niranjand22, changes are done, kindly check and update.

**Comment by niranjand22** [comment]
Hi @Mahesh3ktech 
I tested..blank report is getting generated..not data,no column, no heading

[Accrual (8).xlsx](https://github.com/3ktech/General/files/11513186/Accrual.8.xlsx)

**Comment by Mahesh3ktech** [comment]
hi @niranjand22 Along with start and end date balance account no. filter is mandatory.
Add the balance account no. and generate the report it will work.

**Comment by niranjand22** [comment]
client has to confirm

**Comment by niranjand22** [comment]
In accrual schedule..they want balance to to be displayed as on date..accural and reversal value is also coming diffrent

**Comment by niranjand22** [comment]
Technical changes requried @Mahesh3ktech .
Opening balance not showing previous month closing balance value

**Comment by Mahesh3ktech** [comment]
hi @niranjand22 changes has been done can we connect today to test the same.

**Comment by niranjand22** [comment]
@Mahesh3ktech  client need open balance clumn to be update for next month data..we will connect on this

**Comment by niranjand22** [comment]
@Mahesh3ktech  let us know once you available..few feedback received by client.need to discuss with you

**Comment by niranjand22** [comment]
@Mahesh3ktech  please update once opening balance value you configure

**Comment by Mahesh3ktech** [comment]
hi @niranjand22, opening balance changes are done, kindly check and update.

**Comment by niranjand22** [comment]
Opening and closing balance is not showing the values correctly,

**Comment by niranjand22** [comment]
Technical is not avaialble for further changess

**Comment by audhi3k** [comment]
@3ktech, need technical support to close this.  This was handled by Mr @Mahesh3ktech

**Comment by niranjand22** [comment]
REport is getting re  developed as per their client requirnment.

**Comment by TabrejAlam3K** [comment]
Hi Mr. @Mahesh3ktech - As per our discussion and show case yesterday, Data is not coming correctly, Please correct the output data and confirm. Based on your confirmation we will forward this to the client for Data Verification testing.

CC : @audhi3k @niranjand22

**Comment by 3ktech** [comment]
hi @TabrejAlam3K , we would like to clarify couple of things in the report. can you please reach out to me as soon as you are available.

**Comment by TabrejAlam3K** [comment]
Hi Mr. @rohit3k - Please update the current status of the development.

CC : @audhi3k @Shasthara @narayana3kt

**Comment by TabrejAlam3K** [comment]
Time effort Required by the Technical is - 6 Hours (Including technical testing also).

**Comment by TabrejAlam3K** [comment]
![Image](https://github.com/3ktech/General/assets/116171982/8cde2b09-9c64-4a59-ad35-35a0fdd7976b)

**Comment by rohit3k** [comment]
@TabrejAlam3K as per requirement Report has been developed, please check if anything required.

cc @3ktech @audhi3k

**Comment by TabrejAlam3K** [comment]
Thanks Mr. @rohit3k - I have tested and found it looks good. I have forwarded to the client for Unit Testing, If they raise any points regarding this I will highlight to you.

CC : @audhi3k

**Comment by TabrejAlam3K** [comment]
Information passed to - ⁠Pavan Narasimha - Please test the report and confirm.

⁠1. We want all the months to appear in the report and if the values are not there and it shall show us "0" - It is not Possible. - **Not Possible**.

⁠2. Only transactions affecting because of Accrual should be considered and other means like invoice booking, payment, GJ, etc shouldn't affect this report. - **Completed**

3. ⁠If an account is having Accrual balance, then the report can show positive balance and if there is more reversal then it can be negative. - **Completed**
⁠
4. Calculation shall be opening + months balance and not Opening - Months' balance. - **Completed**

Note - If you are not putting any value running the report in the field Accrual GL Account No.  It will auto considered as Accrual (425100) Account.

**Comment by TabrejAlam3K** [comment]
Final UAT Done with the client, He want to review the entries and let us update for confirmation.

CC : @audhi3k @3ktech

**Comment by Shasthara** [comment]
@TabrejAlam3K - any update here?

**Comment by TabrejAlam3K** [comment]
We will close this ticket tomorrow 17-08-2023 due to non response by the client.

CC : @audhi3k @Shasthara

**Comment by TabrejAlam3K** [comment]
Hi Team we are closing this ticket as of now, If any response receive from the client for this we will create a separate ticket for this.

CC : @audhi3k @Shasthara
